jetty-websocket/websocket-core/src/main/java/org/eclipse/jetty/websocket/core/client/ClientUpgradeRequest.java

```diff
@@ -409,8 +409,6 @@ else if (values.length == 1)
 
         // Verify the negotiated subprotocol
         List<String> offeredSubProtocols = getSubProtocols();
-        if (negotiatedSubProtocol == null && !offeredSubProtocols.isEmpty())
-            throw new WebSocketException("Upgrade failed: no subprotocol selected from offered subprotocols ");
```

Comment:
So it is OK to not select a subprotocol?

Reply:
It's okay to not select one of the offered subprotocols as long as no subprotocol header is sent back.

```
If the client's handshake did not contain such a header field or if
the server does not agree to any of the client's requested
subprotocols, the only acceptable value is null.  The absence
of such a field is equivalent to the null value (meaning that
if the server does not wish to agree to one of the suggested
subprotocols, it MUST NOT send back a |Sec-WebSocket-Protocol|
header field in its response).
```